octopus.js

```diff
@@ -105,17 +113,21 @@ module.exports = function(RED) {
 
                                 var msg3 = {};
                                 msg3.payload = [];
-
-                                msg.payload.results.forEach(function(item, index) {
-                                    msg3.payload.push([{ value_inc_vat : item.value_inc_vat, 
+								if (APIurl.includes("OUTGOING")) {
```

Comment:
Why not just use the `this.tariff == "OUTGOING"` comparison?

Reply:
I tried this, but I think "this" doesn't refer to the node by this point in the function (and therefore this.tariff doesn't evaluate to anything)? The condition always evaluated to false. I couldn't think of a way to get the tariff value at that point in the code without just using the APIurl which I knew was there. I'm not massively familiar with js, so it may just be me not understanding.

Reply:
This was the final 2 commits, the penultimate one was verifying what I thought was happening (both outputs swapped to Agile), and the final commit let them evaluate to different things.

Reply:
Looking at this (and to be honest my JS is a little minimal) I wonder why I ever moved it from `n.tariff` to `this.tariff`. `n.tariff` should be fine to use throughout. Try that.

Reply:
> Looking at this (and to be honest my JS is a little minimal) I wonder why I ever moved it from `n.tariff` to `this.tariff`. `n.tariff` should be fine to use throughout. Try that.

Would you prefer I change this.region as well, or just this.tariff?

Reply:
That was my point. If you set a variable for `source` dependent on the tariff in use, you can use it without resorting to multiple 'if' statements and repeating code blocks.